packages/block-library/src/post-comments/edit.js

```diff
@@ -18,19 +28,56 @@ function PostCommentsDisplay( { postId } ) {
 			// TODO: "No Comments" placeholder should be editable.
 			return comments && comments.length
 				? comments.map( ( comment ) => (
-						<p key={ comment.id }>{ comment.content.raw }</p>
+						<p key={ comment.id }>
+							<RawHTML>{ comment.content.raw }</RawHTML>
+						</p>
 				  ) )
 				: __( 'No comments.' );
 		},
 		[ postId ]
 	);
 }
 
-export default function PostCommentsEdit() {
-	// TODO: Update to handle multiple post types.
-	const postId = useEntityId( 'postType', 'post' );
-	if ( ! postId ) {
-		return __( 'Post Comments' );
+export default function PostCommentsEdit( {
+	attributes,
+	setAttributes,
+	context,
+} ) {
+	const { postType, postId } = context;
+	const { textAlign } = attributes;
+
+	if ( ! postType || ! postId ) {
+		return (
+			<Warning>{ __( 'Post comments block: no post found.' ) }</Warning>
+		);
+	} else if ( postType !== 'post' ) {
+		return (
+			<Warning>
+				{ __(
+					'Post comments block: Comments are only available in posts. Please add this block to a post instead.'
```

Comment:
I don't really agree with this, as it's not strictly true.

I'd rather we checked if the post type (or even the post!) supports comments, than just relying on this check.

Reply:
I removed this extra check for now, so currently it will just fall back to 'No comments.' in that odd case.

> I'd rather we checked if the post type (or even the post!) supports comments

That sounds like a good way to do this!  Although I'm not sure how to go about checking that currently and it may be best to wait as it seems like this block will require some significant restructuring in the future (#24101).

Reply:
Oh yes definitely!
Right now I see this block as almost like a placeholder for the comments loop, so I think it's reasonable to keep chiseling it into shape.

The feature support check should be a major concern for when we move to the multipage editing, especially considering how extensible _everything_ in WP is.
Anything could be added or removed from any post types, or even posts (e.g. comments can be disabled on specific posts with the Allow Comments toggle in the document sidebar).
But, this affects almost all FSE blocks, so we'll need to figure out a proper solution that fits most/all use cases.

tl;dr this PR is not the most appropriate place to figure it out. 🙂